batch_job/src/work_queue_factory.c

```diff
@@ -1248,21 +1271,30 @@ int main(int argc, char *argv[])
 		return 1;
 	}
 
-	char cmd[1024];
+	char* cmd;
 	if(worker_command != NULL){
-		sprintf(cmd,"cp '%s' '%s'",worker_command,scratch_dir);
+		cmd = string_format("cp '%s' '%s'",worker_command,scratch_dir);
 		if(system(cmd)){
 			fprintf(stderr, "work_queue_factory: Could not Access specified worker_queue_worker binary.\n");
 			exit(EXIT_FAILURE);
 		}
 	}else{
-		sprintf(cmd,"cp \"$(which work_queue_worker)\" '%s'",scratch_dir);
+		cmd = string_format("cp \"$(which work_queue_worker)\" '%s'",scratch_dir);
 		if (system(cmd)) {
 			fprintf(stderr, "work_queue_factory: please add work_queue_worker to your PATH.\n");
 			exit(EXIT_FAILURE);
 		}
 	}
 
+	if(runos_os) {
+		cmd = string_format("cp \"$(which vc3-builder)\" '%s'", scratch_dir);
```

Comment:
free(cmd), since it was allocated in the if-else block above.

Reply:
Done! Also changed it in other places to reflect that change.